src/cli/commands/install.js

```diff
@@ -660,7 +660,7 @@ export class Install {
     }
 
     // build lockfile location
-    const loc = path.join(this.config.cwd, constants.LOCKFILE_FILENAME);
+    const loc = path.join(this.config.worktreeFolder || this.config.cwd, constants.LOCKFILE_FILENAME);
```

Comment:
Here are the options (I assume in these option that there is a `workspaceFolder` variable, but it work the same way if we don't):

1. Keep things as they are (eh), with the worktree definition being "the closest location where lies the `workspaces` field", and explicitely specify the locations from which we want to read the lockfile

2. Redefine `worktreeFolder` as being "the closest location where lies the lockfile", which would effectively `worktreeFolder || workspaceFolder || cwd` - the reason I didn't do this is that it means that a project with no workspaces would *internally* work just like a project with a single workspace (we wouldn't be able to distinguish a project that doesn't use workspaces like we currently can with `worktreeFolder === null`). That's not a problem strictly speaking, but we haven't specified this in our RFCs so we have to agree on this before I can do this.

3. Add a new variable (let's say `lockfileFolder`) that would be defined as `worktreeFolder || workspaceFolder || cwd`. The only issue I see with this is that it would mean yet another folder variable.

What do you think? My order of preference is 3 > 1 > 2